### PR TITLE
Added vars to postgresql module

### PR DESCRIPTION
**Pull request by michaeldisaro**
### List of changes
Added vars:
- replica_enabled
- geo_redundant_backup_enabled
- high_availability_enabled
- auto_grow_enabled

Used these vars to enables replica and backup features instead of using tier to calculate them.

### Motivation and context
We need to explicitly enabled backup and replica features instead of having them calculated from the tier.

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### CODE REVIEW

**Review by Krusty93 — COMMENTED** [review]
What is blocking/concerning you to set the tier with required features?

**Comment by michaeldisaro** [comment]
> What is blocking/concerning you to set the tier with required features?

Something is simply not working in ITN, for example backup geo-redundancy, and something is not wanted, for example the replica. 
These vars allows for more granularity and choice.

**Comment by gunzip** [comment]
We must constraint granularity and choices to predefined tiers anyway.

**Review by gunzip — CHANGES_REQUESTED** [review]
We must constraint granularity and choices to predefined tiers anyway.

**Comment by michaeldisaro** [comment]
> We must constraint granularity and choices to predefined tiers anyway.

Ok, so I'll migrate to ITN in QX 2025+ to wait for geo-redundant backups in ITN. That's fine.

**Comment by michaeldisaro** [comment]
> We must constraint granularity and choices to predefined tiers anyway.

https://github.com/pagopa/dx/pull/175/commits/f5aca60106ab48b3923570f2060353930b7eef3b

File: infra/modules/azure_postgres_server/variables.tf [hunk 1]
```diff
@@ -101,6 +101,30 @@ variable "replica_zone" {
...
+variable "replica_enabled" {
+  type        = bool
+  description = "(Optional) Enables replica for PostgreSQL Flexible Server."
+  default     = false
+}
+
+variable "geo_redundant_backup_enabled" {
+  type        = bool
+  description = "(Optional) Enables geo redundancy for PostgreSQL Flexible Server backups."
+  default     = false
```
Comment (gunzip) [inline infra/modules/azure_postgres_server/variables.tf:113]: this is a breaking change if you put 'false' here. should be opt-out to keep consistent behavior

**Review by gunzip — CHANGES_REQUESTED** [review]

File: infra/modules/azure_postgres_server/variables.tf (hunk 1 above)
Comment (michaeldisaro) [reply infra/modules/azure_postgres_server/variables.tf:113]: suggest the change

**Pull request closed without merging**